pkg/alertmanager/alertstore/store.go

```diff
@@ -39,6 +39,16 @@ type AlertStore interface {
 	// DeleteAlertConfig deletes the alertmanager configuration for an user.
 	// If configuration for the user doesn't exist, no error is reported.
 	DeleteAlertConfig(ctx context.Context, user string) error
+
+	// GetFullState loads and returns the alertmanager state for the given user.
+	GetFullState(ctx context.Context, user string) (alertspb.FullStateDesc, error)
+
+	// SetFullState stores the alertmanager state for the given user.
+	SetFullState(ctx context.Context, user string, fs alertspb.FullStateDesc) error
+
+	// DeleteFullState deletes the alertmanager state for an user.
+	// If state for the user doesn't exist, no error is reported.
+	DeleteFullState(ctx context.Context, user string) error
```

Comment:
Are we going to use it? If yes, what's the use case? If not, I would remove it (let's not build logic we don't need - we can do it anytime in the future).

Reply:
I assumed we would need it sooner rather than later to implement tenant clean-up as an extension of the work @pstibrany did - https://github.com/cortexproject/cortex/pull/3900 https://github.com/cortexproject/cortex/pull/3910

Reply:
Good point. You're right, we'll need it. I missed that, thanks!